### PR TITLE
Support  showing search history on SerachForm

### PR DESCRIPTION
**Pull request by WatanabeToshimitsu**
## What?
Support  showing search history on SearchForm

## Why?
便利なので

## Screenshots or videos
SearchHistory

### CODE REVIEW

**Comment by hdl-service** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/dataware-tools/app-common/pull/202#" title="Author self-approved">WatanabeToshimitsu</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=dataware-tools%2Fapp-common).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/dataware-tools/app-common/blob/master/OWNERS)~~ [WatanabeToshimitsu]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: src/components/SearchForm.tsx [hunk 1]
```diff
@@ -29,44 +64,114 @@ export const SearchFormPresentation = ({
...
+      const newSearchHistory = [...prev];
+      if (index !== -1) {
+        newSearchHistory.splice(index, 1);
+        newSearchHistory.unshift(prev[index]);
+      } else {
+        newSearchHistory.unshift(searchText);
+      }
+
+      localStorage.setItem(
+        searchHistoryKey || "",
```
Comment (d-hayashi) [inline src/components/SearchForm.tsx:161]: これって履歴は無限長に残るのでしょうか？  
パフォーマンスにはそれほど影響しないかもしれないですが、
後々のことを考えると配列の長さを制限するとか、削除機能を付けておいたほうが良いかと思います

Comment (WatanabeToshimitsu) [reply src/components/SearchForm.tsx:165]: @d-hayashi 
ユーザーが明示的に履歴を削除する操作をする気がしなかったので、制限する方向で実装しました！どうでしょうか？

**Comment by WatanabeToshimitsu** [comment]
@d-hayashi 
余分な console.log も消しました

**Review by d-hayashi — APPROVED** [review]
LGTM!

**Comment by hdl-service** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/dataware-tools/app-common/pull/202#pullrequestreview-860626392" title="Approved">d-hayashi</a>*, *<a href="https://github.com/dataware-tools/app-common/pull/202#" title="Author self-approved">WatanabeToshimitsu</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=dataware-tools%2Fapp-common).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/dataware-tools/app-common/blob/master/OWNERS)~~ [WatanabeToshimitsu,d-hayashi]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**